preview/Win7Msix/Win7MSIXInstaller/MsixRequest.cpp

```diff
@@ -168,9 +174,17 @@ HRESULT MsixRequest::ProcessAddRequest()
         HandlerInfo currentHandler = AddHandlers[currentHandlerName];
         AutoPtr<IPackageHandler> handler;
         RETURN_IF_FAILED(currentHandler.create(this, &handler));
-        RETURN_IF_FAILED(handler->ExecuteForAddRequest());
-
-        currentHandlerName = currentHandler.nextHandler;
+        HRESULT hr = handler->ExecuteForAddRequest();
+        if (FAILED(hr))
+        {
+            m_msixResponse->SetErrorCode(hr);

```

Comment:
>m_msixResponse [](start = 12, length = 14)

i think when we talked we wanted to pass the response object to the handlers, and have the response object track that it's canceled instead of the msixRequest., and that we'd call a special error handler which is tracked in the HandlerInfo above.